.github/workflows/ci.yaml

```diff
@@ -18,6 +18,22 @@ jobs:
           command: check
           args: --tests
 
+  build_no_std:
+    name: Build on no_std
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          target: thumbv7em-none-eabi
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: build
+          args: --manifest-path tests/test_no_std/Cargo.toml
```

Comment:
Looking at the [job](https://github.com/upsuper/cstr/actions/runs/3106745679/jobs/5057277415), I strongly suspect this is not doing what you expect it to do.

It should probably have `--target thumbv7em-none-eabi` attached here, and it might be good to also include `--verbose` just to confirm how `rustc` is called for each crate.

(I could do it myself if you can allow me to edit your branch.)

Reply:
```suggestion
          args: >-
            --verbose
            --target thumbv7em-none-eabi
            --manifest-path tests/test_no_std/Cargo.toml
```

Reply:
Good catch, thanks; pulled in to the erroneous commit.